packages/babel-parser/src/options.js

```diff
@@ -1,6 +1,7 @@
 // @flow
 
 import type { PluginList } from "./plugin-utils";
+import sourcemap from "source-map";
```

Comment:
Currently it's very easy to provide a single small bundle for `@babel/parser` because it doesn't have any dependency, but this changes it. Do you think that it would be acceptable to take an instance of `SourceMapConsumer` instead of a plain object?

Reply:
Makes sense. (The best case for me would actually be something sourcemap-library independent, since only the function `(generatedLoc) => sourceloc` is really used).

This also means that automatically recognizing the sourcemap comment wouldn't work, but leaving that to the caller of the parser isn't too bad either (@overlookmotel).

Reply:
@nicolo-ribaudo Would you be ok with accepting a function `({line: number, column: number}) => {line: number, column: number}` as an input sourcemap parameter instead of a `SourceMapConsumer`?

Reply:
Just to throw in my thoughts: While I can see the reason behind not wanting to add dependencies to the parser, personally I think it'd be pretty odd from a user's perspective to have to provide anything other than a source map.

I mean you can imagine the confusion - there's a option called `inputSourceMap` for both `parse()` and `transform()`, but they expect different input.

In fact, it was exactly this confusion which caused me to open #11960 - I saw the `inputSourceMap` option listed in the docs for Babel core/transform and assumed that option would be present on the parser too.

Is there any way around the block on parser having dependencies?

Reply:
I don't think this requires the `source-map` dependency to perform (I'm biased against it ever since they switched to WASM). The algorithm here is extremely simple, we're trying to see if our `loc { lline, column }` matches a mapping from the sourcemap. 

There are _much_ smaller libs (eg, [sourcemap-codec](https://github.com/rich-harris/sourcemap-codec)) which would allow us to decode a sourcemap. The [datastructure](https://github.com/Rich-Harris/sourcemap-codec/blob/master/src/sourcemap-codec.ts#L1-L6) it returns is a little hard (a triple array!), but the alg is essentially:

```js
const line = decoded[parsedLine];
const segment = line.find(segment => {
  return segment[0] === parsedColumn;
});
if (segment) {
  // found a match!
  const filename = map.sources[segment[1]];
  const line = segment[2];
  const column = segment[3];
  const identifierName = map.names[segment[4]];
} else {
  // No match found
}

Reply:
I don't think sourcemap lookup should be reimplemented here. We are probably trading build size for performance and at least your code snippet doesn't handle non exact mappings (e.g. where you need to take an mapping with a lower column and add the difference to parsedColumn).

I think passing a `SourceMapConsumer` is a better solution, but that's also more cumbersome because you need to the parse the sourcemap comment yourself and so on...